crates/net/eth-wire-types/src/types.rs

```diff
@@ -0,0 +1,187 @@
+//! Types and traits for networking types abstraction
+
+use alloy_rlp::{Decodable, Encodable};
+use reth_consensus::ConsensusError;
+use reth_primitives::{
+    alloy_primitives::{Sealable, Sealed},
+    GotExpected, Header, SealedBlock,
+};
+use std::{fmt::Debug, hash::Hash};
+
+/// A trait defining types used in networking.
+pub trait NetworkTypes: Send + Sync + Debug + Clone + Unpin + 'static {
+    /// The block type. Expected to hold header and body.
+    type Block: Block<Header = Self::BlockHeader>;
+    /// The block body type.
+    type BlockBody: BlockBody;
+    /// The block header type.
+    type BlockHeader: BlockHeader;
+
+    /// Helper conversion method building [`SealedBlock`] from sealed header and body.
+    fn build_sealed_block(header: Sealed<Self::BlockHeader>, body: Self::BlockBody) -> SealedBlock;
+
+    /// Helper conversion method splitting [`SealedBlock`] into sealed header and body.
+    fn split_sealed_block(block: SealedBlock) -> (Sealed<Self::BlockHeader>, Self::BlockBody);
+
+    /// Validates block body against header.
+    fn validate_block(
+        header: &Self::BlockHeader,
+        body: &Self::BlockBody,
+    ) -> Result<(), ConsensusError>;
+}
+
+/// Block
+pub trait Block:
+    Encodable + Decodable + Debug + PartialEq + Eq + Clone + Unpin + Send + Sync
+{
+    /// Block header type.
+    type Header: BlockHeader;
+
+    /// Returns block header
+    fn header(&self) -> &Self::Header;
+}
+
+/// Block header
+pub trait BlockHeader:
+    Send
+    + Sync
+    + Encodable
+    + Decodable
+    + Debug
+    + Sealable
+    + Unpin
+    + Clone
+    + PartialEq
+    + Eq
+    + Hash
+    + Into<Header>
```

Comment:
this seems counterproductive if the goal is to remove these extensible types, pls add todo that this is temp hack